CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 2.8.3...3.5)
```

Comment:
The this enough for cmake 4. Don't we need 3.6 or higher?

Reply:
No, CMake 4.0.0 is still compatible with the policy of CMake 3.5 (if you check the error message, it refers to "Compatibility with CMake < 3.5", not "Compatibility with CMake <= 3.5"). This will print a warning for any CMake version >= 3.31, but I wanted to make the least possible invasive change here.

Reply:
I do not think the project is maintained, see https://github.com/wjwwood/serial/issues/275 and https://github.com/wjwwood/serial/issues/312 .

Reply:
fyi I started a friendly fork with (among other things) CMake 4.0 support at https://github.com/ami-iit/serial_cpp, feel free to check it out.